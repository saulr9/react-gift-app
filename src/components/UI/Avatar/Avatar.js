import PropTypes from 'prop-types';
import styles from './Avatar.module.css';

function Avatar({ img }) {
  return (
    <div
      className={styles.avatar}
      style={{
        backgroundImage: `url(${img})`
      }}
    />
  );
}
Avatar.propTypes = {
  img: PropTypes.string.isRequired
};

export default Avatar;
