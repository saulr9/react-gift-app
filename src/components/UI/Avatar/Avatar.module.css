.avatar {
  background-position: center center;
  background-size: cover;
  border-radius: 100%;
  height: 60px;
  width: 60px;
}
