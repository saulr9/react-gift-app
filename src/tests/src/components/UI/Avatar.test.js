import { shallow } from 'enzyme';
import Avatar from '../../../../components/UI/Avatar/Avatar';

describe('Testing UI/Avatar component', () => {
  const img = 'https://www.w3schools.com/w3css/img_lights.jpg';

  const wrapper = shallow(<Avatar img={img} />);

  test('Should render Avatar Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('Div Should have background img', () => {
    const div = wrapper.find('div');
    expect(div.prop('style')).toHaveProperty('backgroundImage', `url(${img})`);
  });
});
