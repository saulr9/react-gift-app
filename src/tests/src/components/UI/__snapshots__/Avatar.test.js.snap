// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing UI/Avatar component Should render Avatar Component 1`] = `
<div
  className="avatar"
  style={
    Object {
      "backgroundImage": "url(https://www.w3schools.com/w3css/img_lights.jpg)",
    }
  }
/>
`;
